packages/remix-dev/vite/plugin.ts

```diff
@@ -91,7 +91,7 @@ const resolveFileUrl = (
   // Vite will prevent serving files outside of the workspace
   // unless user explictly opts in with `server.fs.allow`
   // https://vitejs.dev/config/server-options.html#server-fs-allow
-  if (!isWithinRoot) return `/@fs` + filePath;
+  if (!isWithinRoot) return path.posix.join(`/@fs`, normalizePath(filePath));
```

Comment:
The idea is to just mimic what vite does here

https://github.com/vitejs/vite/blob/b93dfe3e08f56cafe2e549efd80285a12a3dc2f0/packages/vite/src/node/config.ts#L506

```
path.posix.join(FS_PREFIX, normalizePath(ENV_ENTRY))
```